### DISCUSSION TITLE
[WAYLAND SESSION] While using fractional scaling, include option to ask XWayland clients to scale themselves

### DISCUSSION THREAD

**Discussion by Constancies** [linuxmint/discussions#262] [Ideas]
Currently in Cinnamon’s Wayland session, fractional scale factors cause the compositor to manually zoom in XWayland apps from 100% to the desired scale factor, resulting in a blurry image. However, many apps, such as Discord, VSCode, Steam (+ nearly all of its games), and MuseScore support scaling themselves, but don’t have a Wayland session (or it is broken in some way that results in it being disabled by default).

Because of this, it’s possible to pass arguments to these XWayland apps so that they are able to scale themselves without causing ugly blurring. This is an option KDE Plasma created about a year and a half ago, as detailed in this blog post by Nate Graham: https://pointieststick.com/2022/06/17/this-week-in-kde-non-blurry-xwayland-apps/

There is a downside to this: for apps that aren’t able to scale themselves, they look unusably small.  Because of that, If something like this is implemented, I propose that, unlike Plasma, this option is disabled by default, since it may not be worth new users encountering a really tiny app on a fractional scaled screen and struggling to use it. However, since there are a significant amount of apps stuck in X11 for one reason or another for the foreseeable future, and many of them do have built-in scaling, it makes sense to include an option like this for advanced users who wish to get the most out of their hidpi screens.

**Comment by anaximeno** [comment]
I believe this can be reported here https://github.com/linuxmint/wayland
There might be better to track while making other fixes for wayland.

**Reply by Constancies** [reply]
Alright, I'll go ahead and repost it there ^^
